### PR TITLE
[ringbuffer] ringbuffer size boost to 32bit

### PR DESCRIPTION
**Pull request by Z8MAN8**
## 拉取/合并请求描述：(PR description)

[

#### 为什么提交这份PR (why to submit this PR)
将ringbuffer支持的长度改为32位

related: https://github.com/RT-Thread/rt-thread/issues/682

#### 在什么测试环境下测试通过 (what is the test environment)

]

### 当前拉取/合并请求的状态 Intent for your PR

必须选择一项 Choose one (Mandatory):

- [ ] 本拉取/合并请求是一个草稿版本 This PR is for a code-review and is intended to get feedback
- [ ] 本拉取/合并请求是一个成熟版本 This PR is mature, and ready to be integrated into the repo

### 代码质量 Code Quality：

我在这个拉取/合并请求中已经考虑了 As part of this pull request, I've considered the following:

- [ ] 已经仔细查看过代码改动的对比 Already check the difference between PR and old code
- [ ] 代码风格正确，包括缩进空格，命名及其他风格 Style guide is adhered to, including spacing, naming and other styles
- [ ] 没有垃圾代码，代码尽量精简，不包含`#if 0`代码，不包含已经被注释了的代码 All redundant code is removed and cleaned up
- [ ] 所有变更均有原因及合理的，并且不会影响到其他软件组件代码或BSP All modifications are justified and not affect other components or BSP
- [ ] 对难懂代码均提供对应的注释 I've commented appropriately where code is tricky
- [ ] 代码是高质量的 Code in this PR is of high quality
- [ ] 已经使用[formatting](https://github.com/mysterywolf/formatting)等源码格式化工具确保格式符合[RT-Thread代码规范](https://github.com/RT-Thread/rt-thread/blob/master/documentation/contribution_guide/coding_style_cn.md) This PR complies with [RT-Thread code specification](https://github.com/RT-Thread/rt-thread/blob/master/documentation/contribution_guide/coding_style_en.md)

### CODE REVIEW

**Review by mysterywolf — APPROVED** [review]

File: components/drivers/ipc/ringbuffer.c [hunk 1]
```diff
@@ -157,8 +157,7 @@ rt_size_t rt_ringbuffer_put_force(struct rt_ringbuffer *rb,
 
     if (length > space_length)
     {
-        if (rb->write_index <= rb->read_index)
-            rb->read_mirror = ~rb->read_mirror;
```
Comment (mysterywolf) [inline components/drivers/ipc/ringbuffer.c:161]: 此处需要确认一下

Comment (mysterywolf) [reply components/drivers/ipc/ringbuffer.c:161]: 这个地方被revert回去了 不应该删掉的：https://github.com/RT-Thread/rt-thread/pull/4904/commits/2439232c8a81a906bc119f12ed86bada0ed4df42

**Review by mysterywolf — APPROVED** [review]

**Review by liu2guang — APPROVED** [review]
+1

**Comment by mysterywolf** [comment]
@Guozhanxin 请squash合并

File: components/drivers/include/ipc/ringbuffer.h [hunk 2]
```diff
@@ -38,17 +38,17 @@ struct rt_ringbuffer
      * +---+---+---+---+---+---+---+|+~~~+~~~+~~~+~~~+~~~+~~~+~~~+
      * read_idx-^ ^-write_idx
      *
-     * The tradeoff is we could only use 32KiB of buffer for 16 bit of index.
+     * The tradeoff is we could only use 64KiB of buffer for 32 bit of index.
```
Comment (armink) [inline components/drivers/include/ipc/ringbuffer.h:41]: 这里是不是 2的31 次方，不是 64KB 呢

Comment (mysterywolf) [reply components/drivers/include/ipc/ringbuffer.h:41]: 是的~ 应该是2^31

File: components/drivers/include/ipc/ringbuffer.h [hunk 3]
```diff
@@ -38,17 +38,17 @@ struct rt_ringbuffer
...
      * But it should be enough for most of the cases.
      *
      * Ref: http://en.wikipedia.org/wiki/Circular_buffer#Mirroring */
```
Comment (mysterywolf) [inline components/drivers/include/ipc/ringbuffer.h:44]: 这段注释可以删掉了，语义已经不再适用了
```suggestion
```

**Review by mysterywolf — APPROVED** [review]

**Review by armink — APPROVED** [review]

**Pull request merged**